blocks.js

```diff
@@ -174,6 +166,19 @@ module.exports = function (file, block_size, cache) {
         }
       })
     },
+    /**
+     * Writes a buffer directly to a position in the file.
+     * This wraps `file.write()` and removes the block cache.
+     *
+     * @param {buffer} buf - the data to write to the file
+     * @param {number} pos - position in the file to write the buffer
+     * @param {function} cb - callback that returns any error as an argument
+     */
+    write: (buf, pos, cb) => {
+      const i = Math.floor(pos/block_size)
+      cache.remove(i)
```

Comment:
> I don't see anything that clears the cached block

@dominictarr did you see this bit? I *think* this clears the cached block, but it's possible I'm doing it wrong.

Reply:
oh, that looks right! you should do it after the write succeeds though, in case a read gets in and sets that item before you callback. hmm... really reads to that item should be blocked while you are writing...